AppCenter/AppCenter/Internals/Util/MSUtility+File.m

```diff
@@ -24,16 +24,14 @@ + (NSURL *)createFileAtPathComponent:(NSString *)filePathComponent
     if (filePathComponent) {
       NSURL *fileURL = [[self appCenterDirectoryURL] URLByAppendingPathComponent:filePathComponent];
 
-      // Check if item already exists.
+      // Check if item already exists. We need to check this as writeToURL:atomically: can override an existing file.
       if (!forceOverwrite && [fileURL checkResourceIsReachableAndReturnError:nil]) {
         return fileURL;
       }
 
       // Create parent directories as needed.
       NSURL *directoryURL = [fileURL URLByDeletingLastPathComponent];
-      if (![directoryURL checkResourceIsReachableAndReturnError:nil]) {
-        [self createDirectoryAtURL:directoryURL];
-      }
+      [self createDirectoryAtURL:directoryURL];
```

Comment:
Should we check the return value and log any errors here? Your comment mentions that **it is more efficient to attempt the operation and handle any failure that may occur.** which means we should handle errors in each operation.

Reply:
That's true, the issue here is that we can't really do anything except log an error message.
I've committed a change that handles the return value of createDirectoryAtURL and that tries to be more verbose in code as to what's going on.
Now about the actual error message of `createDiretoryAtURL:withIntermediateDirectories:attributes:error:`: we log the error message.
I can't come up with a more useful logic for errors. I was considering shutting down the SDK as we are useless in case the SDK cannot create any directories but shutting down the SDK seemed too radical.
Opinions?

Reply:
@jaelim-ms no idea why github is hiding my comment.

Reply:
Probably don't need to shutdown the SDK, it will add complexity for not that much since the SDK won't work anyway. It should just not crash if such case happens.